app/code/Magento/CatalogInventory/Model/Indexer/Stock.php

```diff
@@ -54,6 +54,7 @@ public function __construct(
     public function execute($ids)
     {
         $this->_productStockIndexerRows->execute($ids);
+        $this->getCacheContext()->registerEntities(ProductModel::CACHE_TAG, $ids);
```

Comment:
This implementation is cleanup all caches in case on  changes to a inventory, not only in. case of inventory status.
As result product page/category page cache will be cleanup after any checkout and significanly reduce cache hit/miss

Reply:
It's the main idea. We need to update product page in case ANY inventory change for product, not only status but also quantity.
ERP updates inventory using API and we need product pages updated accordingly.
Maybe its not suitable for Magento platform itself, in this case just close this.